### ISSUE TITLE
When setting an embargo date to a bitstream, DSpace set the date to the day before not the one entered.

### ISSUE THREAD

**Issue by pilasou** [DSpace/DSpace#10119]
## Describe the bug
When adding an embargo policy to a bitstream of an archived item in DSpace, the date used is the day before not the one entered (if the embargo end date is 2025-12-31, DSPace will store the date 2025-12-30).

## To Reproduce
1. In DSpace send box (https://demo.dspace.org/), connect as admin and find a publication item.
2. On the item page, clic on the edit button (top right)
3. Go to the Bitstream tab
4. In the ORIGINAL Bundle, edit a file
5. Clic on _Edit bitstream's Policies_ link at the bottom of the page.
6. Edit the policy of the Anonymous group and enter a Start Date for the embargo (i.e. 2025-12-31). 
7. Clic the Save button.
8. Refreh the policy page. 
9. The Start date will be 2025-12-30 not 2025-12-31.

## Expected behavior
The date must be the same as the one entered.

**Comment by tdonohue** [comment]
This sounds very similar to an older bug we had DSpace/dspace-angular#1256.  In that case, the bug was caused by a changing the timezone of the date (as the backend always uses UTC as a timezone).  I had thought we had fixed that, but maybe it has popped up again.

Needs a volunteer to investigate why this is occurring again.

**Comment by autavares-dev** [comment]
Tested on a DSpace in a UTC-03 timezone. If the front-end sends, for example, "2025-02-20T00:00:00Z" (which is UTC+00), that value will be processed in:

https://github.com/DSpace/DSpace/blob/d76bdc6d476fa07ce518fa920dd0b1559228171f/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/resourcePolicy/ResourcePolicyStartDateReplaceOperation.java#L61 

and converted to the default timezone (`TimeZone.getDefault()`), correctly resulting in `Wed Feb 19 21:00:00 BRT 2025`. 
However, when stored in the database (column 'start_date' of type 'date'), the stored value is '2025-02-19', it is not being converted back to UTC. 

`TimeZone.getDefault()` is returning for me "America/Sao_Paulo", and this setting is not defined anywhere in the project files. 
It looks like Spring is using the host OS's timezone as default (changed mine to check). 

In the database, `show timezone` returns "America/Sao_Paulo". 

After adding this bean to force UTC as default (there are [other options](https://www.baeldung.com/spring-boot-set-default-timezone) to achieve the same result), the issue with embargo dates was resolved.

```java
@Component
public class GlobalTimezoneBeanFactoryPostProcessor implements BeanFactoryPostProcessor {
    @Override
    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
    }
}
```

I can't tell how this could impact other parts of the system and values already stored in the database. I modified an item to check the 'last_modified' value, and it was still stored correctly.

**Comment by tdonohue** [comment]
All, I've just realized this bug ticket is likely fixed by #10432 .  In that larger PR, we migrated all dates to `java.time.*`, but it also **included** a setting very similar to what @autavares-dev [notes above](https://github.com/DSpace/DSpace/issues/10119#issuecomment-2672273847).   

See the changes made to `org.dspace.app.rest.WebApplication` made in this commit https://github.com/DSpace/DSpace/pull/10432/commits/837f7c993cad9accef3f0493c6385e101d8b812d   Those fixes are essentially the same as what @autavares-dev proposed above.

At this point in time this is ONLY fixed in 9.0 (upcoming release).  However, we could consider backporting those changes to `WebApplication` to 8.x if we find that small fix is necessary for 8.x.

**Comment by kshepherd** [comment]
Note, though, that #10432 doesn't seem to change `DateMathParser` which is used in submission access condition step and in the REST convert class - it is still using `Date` object, and does some local / timezone specific stuff that might be relevant

**Comment by tdonohue** [comment]
@kshepherd : I realized I didn't clarify my prior message. To clarify, I'm not completely convinced that the bug is in `DateMathParser`.  At least in my work in #10432, I feel I narrowed this down to being an issue in either Hibernate or Spring **after reading the date from the database** (though it is possible that #10432 fixed some issues with storing the date properly and only encountered issues with reading the data properly)

So, from what I saw, it looked like the date was **stored correctly** in the database..  The issue that I saw was that Hibernate or Spring (haven't figured out which, though I suspect Hibernate) will **automatically convert dates to your local timezone when they are read from the database**.

As an example, you might set an embargo for `2026-01-31`.  I was finding it would be stored in the database table correctly as `2026-01-31`, but when it is read from the database, it *would sometimes be* read as `2026-01-30` because of a difference in timezone between the database (storing everything in UTC) and your JVM.  Whatever is reading the date would essentially read `2026-01-31` as if it was `2026-01-31T00:00:00Z`, and then transform it to your local timezone (which might be UTC-3) as `2026-01-30T21:00:00Z`, which would be truncated back to `2026-01-30` (one day before). 

The only fix I could find was to set the default timezone to UTC, as done in https://github.com/DSpace/DSpace/commit/837f7c993cad9accef3f0493c6385e101d8b812d (see the changes to `WebApplication`).  Obviously this would avoid that auto-conversion of dates because it ensures the JVM timezone is UTC just like the database timezone.

**Comment by tdonohue** [comment]
I believe this is now fixed as of DSpace 9.0 (likely fixed by #10432).  However, that needs verification.

**Comment by pilasou** [comment]
Hi @tdonohue, as the fixed has been merged, does this mean I can test on DSpace Sandbox website (https://sandbox.dspace.org/home) ? Just to be sure it solved this problem.

**Comment by tdonohue** [comment]
@pilasou : Yes, the fix should be deployed to both demo.dspace.org and sandbox.dspace.org.  I *believe* this bug should now be fixed, but I'd definitely appreciate a verification test if you have time.

**Comment by pilasou** [comment]
@tdonohue I have tested on both sandbox and demo sites. Happy to say that it worked! The date stays the same. I modified an existing bitstream policy in the Original bundle (the policy was READ for the anonymous user group, not dates were set). I added an embargo until 2025-07-31. Just for reference, following are 2 print screens: 

**Embargo policy set up**

<img width="1282" height="354" alt="Image" src="https://github.com/user-attachments/assets/e65e3cf6-4f0e-4900-b704-e246ab40bcdd" />

**Results once saved** 

<img width="1312" height="240" alt="Image" src="https://github.com/user-attachments/assets/6b45a7e6-5ffd-440d-b245-0c15bb5ba1ca" />

 This issue can be closed.

**Comment by tdonohue** [comment]
I believe this is now also fixed on `dspace-8_x` (via #11111) and `dspace-7_x` (via #11113) branches.  So, this bug should be resolved also in v7.6.6 and v8.3 (once those are released), along with 9.0.  However, verification testing on the latest `dspace-7_x` and `dspace-8_x` code is welcome.
